utils_nlp/models/transformers/named_entity_recognition.py

```diff
@@ -1,2 +1,430 @@
+            weight_decay=weight_decay,
+            learning_rate=learning_rate,
+            adam_epsilon=adam_epsilon,
+            warmup_steps=warmup_steps,
+            verbose=verbose,
+            seed=seed,
+        )
+
+
+    def predict(
+        self,
+        eval_dataset,
+        device="cuda",
+        batch_size=32,
+        num_gpus=None,
+        local_rank=-1,
+        verbose=False,
+    ):
+        """
+        Test on an evaluation dataset and get the token label predictions.
+
+        Args:
+            eval_dataset (TensorDataset): A TensorDataset for evaluation.
+            device (torch.device, optional): A PyTorch device.
+                Defaults to 'cuda'.
+            batch_size (int, optional): The batch size for evaluation.
+                Defaults to 32.
+            num_gpus (int, optional): The number of GPUs to be used.
+                Defaults to None, all gpus are used.
+            local_rank (int, optional): Whether need to do distributed training.
+                Defaults to -1, no distributed training.
+            verbose (bool, optional): Verbose model.
+                Defaults to False.
+        
+        Returns:
+            Numpy ndarray of raw predictions. The shape of the ndarray is
+            [number_of_examples, sequence_length, number_of_labels]. Each
+            value in the ndarray is not normalized. Post-process will be needed
+            to get the probability for each class label.
+        """
+
+        preds = list(
+            super().predict(
+                eval_dataset=eval_dataset,
+                get_inputs=TokenClassificationProcessor.get_inputs,
+                device=device,
+                per_gpu_eval_batch_size=batch_size,
+                n_gpu=num_gpus,
+                local_rank=local_rank,
+                verbose=verbose,
+            )
+        )
+        preds_np = np.concatenate(preds)
+        return preds_np
+
+
+    def get_predicted_token_labels(
+        self,
+        predictions,
+        label_map,
+        dataset
+    ):
+        """
+        Post-process the raw prediction values and get the class label for each token.
+
+        Args:
+            predictions (ndarray): A numpy ndarray produced from the `predict` function call.
+                The shape of the ndarray is [number_of_examples, sequence_length, number_of_labels].
+            label_map (dict): A dictional object to map a label (str) to an ID (int). 
+                dataset (TensorDataset): The TensorDataset for evaluation.
+
+        Returns:
+            List of lists. The size of the retured list is the number of testing samples.
+                Each sublist represents the predicted label for each token. 
+        """
+
+        num_samples = len(dataset.tensors[0])
+        if num_samples != predictions.shape[0]:
+            raise ValueError("Predictions have {0} samples, but got {1} samples in dataset".format(predictions.shape[0], num_samples))
```

Comment:
This line is longer than 100. Are you using the black and flake8 git hooks for autoformatting?

Reply:
Done.